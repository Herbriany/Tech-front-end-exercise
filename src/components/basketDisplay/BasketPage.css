.basket-page {
  max-width: 960px;
  margin-right: auto;
  margin-left: auto;
}

.basket-table-header {
  border-bottom: 1px solid #6C757D;
  padding-bottom: 10px;
}

.basket-product {
  padding: 15px;
  border-bottom: 1px solid #6C757D;
}

.basket-product__img {
  max-height: 80px;
  width: auto;
  padding-right: 5px;
}

.basket-product__amount {
  width: 20px;
  padding: .375rem .75rem;
  margin: auto;
}

.trash-icon {
  height: 20px;
  width: 20px;
  margin-left: 10px;
}

.trash-icon:hover {
  cursor: pointer;
}
/*# sourceMappingURL=BasketPage.css.map */