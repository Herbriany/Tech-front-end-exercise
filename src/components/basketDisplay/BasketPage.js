import React from 'react'
import PropTypes from 'prop-types'

function BasketPage({basketProducts, totalPrice}) {
    return (
        <div className="basketPage">
            <h2>Basket Page</h2>
            <div className="row"> 
                <div className="col col-lg-9 col-sm-12">
                    <div className="row">
                        <div className="col col-lg-10">
                        Items
                        </div>
                        <div className="col col-lg-2">
                        Quantity/Price
                        </div>
                    </div>
                    { basketProducts.map(product =>{ 
                        return ( 
                            <div className="row" key={product.productId}>
                                <div className="col col-lg-10">
                                    <img src={product.image} />
                                </div>
                                <div className="col col-lg-2">
                                    &pound;{product.price}
                                </div>
                            </div>
                            )
                        })  
                    }
                </div>
                <div className="col col-lg-3 col-sm-12">
                    Order summary
                    <div className="row">
                    Total: &pound;{totalPrice}
                    </div>
                </div>
            </div>
        </div>
    )
}

BasketPage.propTypes = {
    basketProducts: PropTypes.array,
    totalPrice: PropTypes.number
}
export default BasketPage