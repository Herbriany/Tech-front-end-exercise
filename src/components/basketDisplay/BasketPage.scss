@import "../../assets/scss/_variables";

.header--basket {
    font-size: 2rem;
}

.basket-page {
    max-width: 960px;
    margin-right: auto;
    margin-left: auto;
}

.basket-table-header {
    border-bottom: 1px solid $grey-border;
    padding-bottom: 10px;
}

.basket-product {
    padding: 15px;
    border-bottom: 1px solid $grey-border;

    &__img {
        max-height: 80px;
        width: auto;
        padding-right: 5px;
    }
    &__amount {
        width: 20px;
        padding: .375rem .75rem;
        margin: auto;
    }

    &__title {
        display: inline;
    }

    &__trash-button {
        border: 0;
        padding: 0;
        background-color: $background-color;
    }

    &__trash-icon {
        height: 20px;
        width: 20px;
        margin-left: 10px;
    
        &:hover {
            cursor: pointer;
        }
    }
}

.basket-summary{
    &__title {
        margin: auto;
    }
    &__total {
        margin: auto;
    }
}

.basket-list {
    &__empty-item {
        margin-top: 10px;
    }
}

.container--empty-basket-button {
    margin: 20px;
}

