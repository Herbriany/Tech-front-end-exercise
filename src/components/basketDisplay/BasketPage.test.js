import React from 'react'
import BasketPage from './BasketPage'
import renderer from 'react-test-renderer'
import { products } from '../../../tools/mockData'
import { shallow } from 'enzyme'

function render(args) {
    const defaultProps = {
        basketProducts: [],
        totalPrice: 0,
        productAmount: 0,
        onDecrementClick: () => {},
        onIncrementClick: () => {},
        onEmptyBasketClick: () => {},
        onRemoveProductClick: () => {}
    }
    const props = {...defaultProps, ...args}
    return shallow(<BasketPage {...props}/>)
}

it("Should read 'Your basket is empty' when no products in basket", () => {
    const wrapper = render()
    expect(wrapper.find("h2").text()).toBe("Your Basket")
    
})

describe("snapshot tests", () => {

    it("Should show 8 products and an order summary", () => {
        const tree = renderer.create(<BasketPage 
            basketProducts={products}
            totalPrice={products[0].price}
            productAmount={products.length}
            onDecrementClick={jest.fn()}
            onIncrementClick={jest.fn()}
            onEmptyBasketClick={jest.fn()}
            onRemoveProductClick={jest.fn()}
        />)
        expect(tree).toMatchSnapshot()
    })

    it("Should show empty basket and no order summary", () => {
        const tree = renderer.create(<BasketPage 
            basketProducts={[]}
            totalPrice={0}
            productAmount={0}
            onDecrementClick={jest.fn()}
            onIncrementClick={jest.fn()}
            onEmptyBasketClick={jest.fn()}
            onRemoveProductClick={jest.fn()}
        />)

        expect(tree).toMatchSnapshot()
    })
})


