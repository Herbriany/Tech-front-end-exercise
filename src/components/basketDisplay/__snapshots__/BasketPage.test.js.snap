// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests Should show 8 products and an order summary 1`] = `
<div
  className="basketPage"
>
  <h2
    className="basket-header"
  >
    Your Basket
  </h2>
  <div
    className="row"
  >
    <div
      className="col-lg-9 col-sm-12 col-xs-12"
    >
      <div
        className="container"
      >
        <div
          className="row basket-table-header"
        >
          <div
            className="col-lg-10"
          >
            <div
              className="row"
            >
              8
               item
              s
            </div>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              Quantity/Price
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/9805244_R_SET?w=110&h=130"
            />
            Russell Hobbs Pennine Illuminating S Steel Kettle 20444
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/4251824_R_SET?w=110&h=130"
            />
            Challenge White Desk Fan - 12 Inch
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/tuc132404445_R_SET?w=110&h=130"
            />
            Plain Vest Top 12 BLACK
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/4049641_R_SET?w=110&h=130"
            />
            Wilson Roger Federer 27 Inch Adult Tennis Racket
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/5399235_R_SET?w=110&h=130"
            />
            My Little Pony Cutie Mark Rainbow Dash
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/7543412_R_SET?w=110&h=130"
            />
            Sim Free iPhone X 256GB Mobile Phone- Space Grey
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/7067462_R_SET?w=110&h=130"
            />
            Assassin's Creed Origins PS4 Game
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row basket-product"
        >
          <div
            className="col-lg-10"
          >
            <img
              src="https://media.4rgos.it/s/Argos/7426735_R_SET?w=110&h=130"
            />
            Duracell Supreme 750 mAh Rechargeable AAA Batteries - 4 Pack
            <svg
              className="trash-icon"
              onClick={[Function]}
              role="img"
              viewBox="0 0 24 24"
            >
              <path
                d="M8 6l.001-1.51h.147a3.49 3.49 0 0 1 3.342-2.488L12.505 2a3.49 3.49 0 0 1 3.49 3.49v.065L15.987 6H20a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2h4zm2.002 0h3.984v-.019l.01-.463V5.49A1.49 1.49 0 0 0 12.506 4l-1.015.002a1.49 1.49 0 0 0-1.49 1.489V6zM8.5 11a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm5 0a1 1 0 0 1 2 0v6a1 1 0 0 1-2 0v-6zm-9.447-.008a1 1 0 1 1 2 .016l-.055 7.094A1.886 1.886 0 0 0 7.88 20h8.237c1.04 0 1.883-.843 1.883-1.892l-.032-7.103a1 1 0 1 1 2-.01l.032 7.122A3.883 3.883 0 0 1 16.118 22H7.851a3.883 3.883 0 0 1-3.853-3.913l.055-7.095z"
                fill="#b8adad"
                fillRule="nonzero"
              />
            </svg>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                -
              </button>
              <div
                className="basket-product-amount"
              />
              <button
                className="btn btn-secondary"
                onClick={[Function]}
              >
                +
              </button>
            </div>
            <div
              className="row"
            >
              £
              NaN
            </div>
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "marginTop": "20px",
            }
          }
        >
          <button
            className="btn btn-danger"
            onClick={[Function]}
          >
            Empty basket
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-lg-3 col-sm-12 col-xs-12"
    >
      <div
        className="row"
      >
        <p
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          Order summary
        </p>
      </div>
      <div
        className="row"
      >
        <p
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          Total: £
          24.99
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot tests Should show empty basket and no order summary 1`] = `
<div
  className="basketPage"
>
  <h2
    className="basket-header"
  >
    Your Basket
  </h2>
  <div
    className="row"
  >
    <div
      className="col-lg-9 col-sm-12 col-xs-12"
    >
      <div
        className="container"
      >
        <div
          className="row basket-table-header"
        >
          <div
            className="col-lg-10"
          >
            <div
              className="row"
            >
              0
               item
              s
            </div>
          </div>
          <div
            className="col-lg-2"
          >
            <div
              className="row"
            >
              Quantity/Price
            </div>
          </div>
        </div>
        <div
          className="row"
          style={
            Object {
              "marginTop": "10px",
            }
          }
        >
          <p>
            Your basket is empty
          </p>
        </div>
        
      </div>
    </div>
  </div>
</div>
`;
